### PR TITLE
SAML authentication feature in Velociraptor

### PR DESCRIPTION
**Pull request by turekt**
Hi,

I've implemented the SAML authentication feature that was requested in issue #50 .
There is a markdown file in docs/saml folder explaining on how to test this feature along with an explanation what needs to be configured for the SAML authentication to work.

I think that this feature is still a work in progress since the SAML certificates need to be manually generated and set in the configuration and there may be some other changes that need to be implemented in order to consider this feature as complete. Additionally, I've used the SAML library that was proposed (https://github.com/crewjam/saml) and there was an unexpected issue when building with `go run make.go -v linux` that seems to be related to go version 1.12  since it throws the following error:
```
../api/saml.go:69:29: samlMiddleware.GetAuthorizationToken undefined (type *samlsp.Middleware has no field or method GetAuthorizationToken)
../api/saml.go:105:17: samlMiddleware.RequireAccountHandler undefined (type *samlsp.Middleware has no field or method RequireAccountHandler)
``` 
I've observed that the methods are actually present and upgrading go to version 1.13 resolved the issue. I'm not sure why this happens and I'm aware that Velociraptor releases might be built by earlier go versions.

The basic SAML authentication was tested with Simple SAML and Microsoft ADFS. Please let me know your thoughts on the implementation and whether this is satisfactory to be merged or there are changes that need to be made beforehand. If you have ideas on why go 1.12 does not recognize the samlsp.Middleware methods, it would be much appreciated to try and find a fix for go versions prior 1.13.

Thanks!

### CODE REVIEW

**Comment by scudette** [comment]
The broken build happens because the wrong version of saml is picked up by dep. If you are using dep you should add this to the Gopkg.toml:

```
[[constraint]]
  name = "github.com/crewjam/saml"
  revision = "344d075952c9343809f57f4e465504dd5e3068a4"
```

But there is no issue with building as a module (simply clone the directory outside the go path and then type make).

File: docs/saml/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+
+This worked for me, but of course I don't know what exactly you're doing so mileage may vary.
+```
+
+The notes were copied from the following [link](https://github.com/crewjam/saml/issues/5#issuecomment-501328253). 
+With this setup we've observed that the `saml_user_attribute` should be set to ` http://schemas.xmlsoap.org/ws/2005/05/identity/claims/upn`.
+
+## Other
+
+We've observed and issue when building a release with go 1.12. The error that is output by go build is
```
Comment (scudette) [inline docs/saml/README.md]: Is this an issue any more?

**Comment by turekt** [comment]
Fixed and done.

**Comment by scudette** [comment]
Thanks!

**Pull request merged**